src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultFileUriProvider.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Composition;
+using System.IO;
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor
+{
+    [Shared]
+    [Export(typeof(FileUriProvider))]
+    internal class DefaultFileUriProvider : FileUriProvider
+    {
+        private readonly ITextDocumentFactoryService _textDocumentFactory;
+
+        [ImportingConstructor]
+        public DefaultFileUriProvider(ITextDocumentFactoryService textDocumentFactory)
+        {
+            if (textDocumentFactory is null)
```

Comment:
>if (textDocumentFactory is null) [](start = 12, length = 32)

Is this even possible in an importing constructor? Doesn't it fail to compose if the dependencies can't resolve?

Reply:
Not possible in VS; however these types can be constructed in other scenarios such as at test time or in the off chance code is re-used outside of VS (i.e. for a language server).